### PR TITLE
Multi Symbol Read for ChunkStore

### PR DESCRIPTION
**Pull request by bmoscon**
@jamesblackburn @RichardWarfield please take a look. This addresses #394 

Its likely it doesnt 100% work correctly yet - I just wanted to code up a proof of concept and a simple unit test to test out base functionality. I haven't updated doc strings, or written up more complicated test cases with things like date ranges, symbols that do not exist, symbols outside of a date range, etc. 

I will add those tests and update the documentation assuming that this is in the right direction.

I also have not updated / changed tickstore or version store. I assume this is something we'd want in VersionStore but not Tickstore. Let me know @jamesblackburn

### CODE REVIEW

File: arctic/chunkstore/chunkstore.py [hunk 1]
```diff
@@ -239,37 +242,43 @@ def read(self, symbol, chunk_range=None, filter_data=True, **kwargs):
...
         DataFrame or Series
         """
-
         sym = self._get_symbol_info(symbol)
         if not sym:
             raise NoDataFoundException('No data found for %s' % (symbol))
 
-        spec = {SYMBOL: symbol,
-               }
+        # special handling based on a single symbol or multi symbol read
```
Comment (pablojim) [inline arctic/chunkstore/chunkstore.py]: I wonder if it would cleaner to treat all cases as a list - e.g. place a single symbol in a one element list and then have a special case for the return value.

File: arctic/chunkstore/chunkstore.py [hunk 2]
```diff
@@ -239,37 +242,43 @@ def read(self, symbol, chunk_range=None, filter_data=True, **kwargs):
...
+                            (START, pymongo.ASCENDING),
                             (SEGMENT, pymongo.ASCENDING)]
         segment_cursor = self._collection.find(spec, sort=by_start_segment)
 
-        chunks = []
+        chunks = defaultdict(list)
         for _, segments in groupby(segment_cursor, key=lambda x: x[START]):
             segments = list(segments)
-            mdata = self._mdata.find_one({SYMBOL: symbol, START: segments[0][START], END: segments[0][END]})
+            mdata = self._mdata.find_one({SYMBOL: segments[0][SYMBOL],
```
Comment (pablojim) [inline arctic/chunkstore/chunkstore.py]: Is there a possible issue here if segments for multiple symbols have the same start date?
Possibly one for a different PR but I wonder if this grouping could be done in Mongo also?

File: arctic/chunkstore/chunkstore.py (hunk 1 above)
Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: yeah's a good idea - it would clean the code up a lot. I'll make that change and update this

File: arctic/chunkstore/chunkstore.py (hunk 2 above)
Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: yeah thats a case I didn't test out. There might be an issue, but if this isn't a feature we want, I don't want to spend the time to work out all the kinks right away

Comment (pablojim) [inline arctic/chunkstore/chunkstore.py]: I think it would be a very useful feature. Especially in equities where the universe is very large but on a given day we trade a sub set of this universe. These leads to a tension between all equity data in a huge single symbol versus a symbol per equity. With an efficient multi symbol read the symbol per equity becomes more attractive.

I think the multiple symbols with the same start data can be solved with:
` for _, segments in groupby(segment_cursor, key=lambda x: (x[START], x[SYMBOL])):`
(untested!)

Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: it appears to work as-is. perhaps because of the way the collection find is already sorted?

Comment (pablojim) [inline arctic/chunkstore/chunkstore.py]: Yeah reading the docs on itertools groupby this will mostly work. I think there is still an edge case. I'll try make a test case for it.
